compiler/rustc_mir_build/src/build/matches/simplify.rs

```diff
@@ -271,7 +271,8 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             PatKind::Variant { adt_def, args, variant_index, ref subpatterns } => {
                 let irrefutable = adt_def.variants().iter_enumerated().all(|(i, v)| {
                     i == variant_index || {
-                        self.tcx.features().exhaustive_patterns
+                        (self.tcx.features().exhaustive_patterns
```

Comment:
what does this do specifically?

Reply:
The following: if the pattern is `SomeEnum::Variant3` and all the other variants are uninhabited, then we skip the discriminant test. I don't fully understand why we only do this when the feature is on, my guess is someone was overly-conservative.

Reply:
See this: https://github.com/rust-lang/rust/blob/master/tests/ui/closures/2229_closure_analysis/run_pass/multivariant.rs